libs/staging-provider/src/Provider/NewWorkspaceStagingProvider.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\StagingProvider\Provider;
+
+use Keboola\StagingProvider\Provider\Configuration\WorkspaceBackendConfig;
+use Keboola\StorageApi\Components;
+use Keboola\StorageApi\Workspaces;
+
+class NewWorkspaceStagingProvider extends AbstractWorkspaceProvider
+{
+    private StorageApiWorkspace $workspace;
+
+    public function __construct(
+        private readonly Workspaces $workspacesApiClient,
+        private readonly Components $componentsApiClient,
+        private readonly WorkspaceBackendConfig $workspaceBackendConfig,
+        private readonly string $componentId,
+        private readonly ?string $configId = null,
+    ) {
+    }
+
+    protected function getWorkspace(): StorageApiWorkspace
+    {
+        if (empty($this->workspace)) {
```

Comment:
dtto (nullable `$this->workspace` + pryc s `empty`)

Reply:
dal jsem tam isset